File: engine/engine-model/src/main/scala/tools/packet/alliance/AllianceNotice.scala
package tools.packet.alliance

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class AllianceNotice(private var _id: Int, private var _notice: String) extends PacketInput {
  def id: Int = _id

  def notice: String = _notice

  override def opcode(): SendOpcode = SendOpcode.ALLIANCE_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/alliance/AllianceNotice.java
package tools.packet.alliance;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record AllianceNotice(Integer id, String notice) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.ALLIANCE_OPERATION;
   }
}
